akka-stream/src/main/scala/akka/stream/impl/fusing/StreamOfStreams.scala

```diff
@@ -60,15 +62,29 @@ import akka.util.ccompat.JavaConverters._
           case single: SingleSource[T] @unchecked =>
             push(out, single.elem)
             removeSource(single)
+          case iterableSource: IterableSource[T] @unchecked =>
+            handleIterableSource(iterableSource)
           case other =>
             throw new IllegalStateException(s"Unexpected source type in queue: '${other.getClass}'")
         }
       }
 
+      private def handleIterableSource(iterableSource: IterableSource[T]): Unit = {
+        val elements = iterableSource.elements
+        if (elements.knownSize == 0) {
+          removeSource(iterableSource)
+        } else if (elements.knownSize == 1) {
+          push(out, elements.head)
+          removeSource(iterableSource)
+        } else {
+          emitMultiple(out, elements, () => removeSource(iterableSource))
```

Comment:
not run with the SupervisionStrategy of IterableSource

Reply:
This won't work, it will have to be something more clever, emitMultiple will replace the current handlers and emit until end so would only work for breadth = 1 (flatMapConcat), for merge it will turn it into concat.

There will have to be something more clever wrt combining the current open streams with current iterators, in the queue keeping their state across pulls from downstream for non 0-1 size cases.